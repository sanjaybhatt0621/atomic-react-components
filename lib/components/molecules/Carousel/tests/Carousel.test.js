import React from 'react';
import { shallow } from 'enzyme';

import { CarouselVanilla, SlideWrapperVanilla, SlideVanilla } from '../index';

describe('<Carousel />', () => {
  let CarouselComponent = '';
  beforeEach(() => {
    CarouselComponent = shallow(
      <CarouselVanilla>
        <SlideWrapperVanilla>
          <SlideVanilla>TEst 1 </SlideVanilla>
          <SlideVanilla>TEst 3 </SlideVanilla>
          <SlideVanilla>TEst 4 </SlideVanilla>
          <SlideVanilla>TEst 2 </SlideVanilla>
        </SlideWrapperVanilla>
      </CarouselVanilla>
    );
  });

  test('should render correctly', () => {
    expect(CarouselComponent).toMatchSnapshot();
  });

  it('should have slider wrapper as one child only', () => {
    expect(CarouselComponent.find('SlideWrapper')).toHaveLength(1);
  });
});

describe('<SlideWrapper />', () => {
  it('should render correctly', () => {
    const SlideWrapperComponent = shallow(
      <SlideWrapperVanilla>
        <SlideVanilla>TEst 1 </SlideVanilla>
        <SlideVanilla>TEst 3 </SlideVanilla>
      </SlideWrapperVanilla>
    );
    expect(SlideWrapperComponent).toMatchSnapshot();
  });

  it('should have only one slide wrapper', () => {
    const SlideWrapperComponent = shallow(
      <SlideWrapperVanilla>
        <SlideVanilla>TEst 1 </SlideVanilla>
        <SlideVanilla>TEst 3 </SlideVanilla>
      </SlideWrapperVanilla>
    );

    expect(SlideWrapperComponent.find('.carousel-section')).toHaveLength(1);
  });

  it('Children should have unique key attached', () => {
    const SlideWrapperComponent = shallow(
      <SlideWrapperVanilla>
        <SlideVanilla>TEst 1 </SlideVanilla>
        <SlideVanilla>TEst 3 </SlideVanilla>
      </SlideWrapperVanilla>
    );

    expect(SlideWrapperComponent.childAt(0).props().id).toEqual('slide_1');
  });
});

describe('<Slide />', () => {
  let SlideComponent = '';

  beforeEach(() => {
    SlideComponent = shallow(<SlideVanilla>TEst 1 </SlideVanilla>);
  });

  test('should render correctly', () => {
    expect(SlideComponent).toMatchSnapshot();
  });

  it('Should not have unique index prop', () => {
    expect(SlideComponent.childAt(0).props().id).not.toEqual('slide_1');
  });
});

describe('<Dots />', () => {
  let Dots = '';

  beforeEach(() => {
    Dots = shallow(<Dots />);
  });

  test('should render correctly', () => {
    expect(Dots).toMatchSnapshot();
  });

  test('should not have children', () => {
    expect(Dots.props().children).toHaveLength(0);
  });
});
