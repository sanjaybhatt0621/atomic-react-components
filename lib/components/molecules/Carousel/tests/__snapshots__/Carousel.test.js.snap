// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> should render correctly 1`] = `
<section
  className="carousel-wrapper"
>
  <ContextProvider
    value={
      Object {
        "setData": [Function],
        "slides": Array [],
      }
    }
  >
    <ContextProvider
      value={
        Object {
          "currentSlide": 1,
          "setSlideIndex": [Function],
        }
      }
    >
      <SlideWrapper>
        <Slide>
          TEst 1 
        </Slide>
        <Slide>
          TEst 3 
        </Slide>
        <Slide>
          TEst 4 
        </Slide>
        <Slide>
          TEst 2 
        </Slide>
      </SlideWrapper>
    </ContextProvider>
  </ContextProvider>
</section>
`;

exports[`<Slide /> should render correctly 1`] = `
<div
  className="slide"
>
  TEst 1 
</div>
`;

exports[`<SlideWrapper /> should render correctly 1`] = `
<div
  aria-label="carousel"
  className="carousel-section"
>
  <Slide
    id="slide_1"
    key=".0"
  >
    TEst 1 
  </Slide>
  <Slide
    id="slide_2"
    key=".1"
  >
    TEst 3 
  </Slide>
</div>
`;
